virttest/test_setup.py

```diff
@@ -451,6 +451,63 @@ def get_multi_supported_hugepage_size(self):
             raise ValueError("Root hugepage control sysfs directory %s did not"
                              " exist" % self.pool_path)
 
+    def get_kernel_hugepages(self, pagesize):
+        """
+        Get specific hugepages number allocated by kernel at runtime
+
+        read page number from
+        /sys/kernel/mm/hugepages/hugepages-${pagesize}kB/nr_hugepages
+
+        :param pagesize: hugepages size
+        :return: page number
+        """
+        pagedir = "%s/hugepages-%skB" % (self.pool_path, pagesize)
+        pagefile = "%s/nr_hugepages" % pagedir
+
+        if not os.path.isdir(pagedir) or not os.path.isfile(pagefile):
+            raise ValueError("Root hugepage control sysfs directory %s or"
+                             " file %s did not exist" % (pagedir, pagefile))
+        with open(pagefile, "r") as f:
+            hp = f.readline().strip()
+        return int(hp)
+
+    def set_kernel_hugepages(self, pagesize, pagenum):
+        """
+        Let kernel allocate some specific hugepages at runtime
+
+        write page number to
+        /sys/kernel/mm/hugepages/hugepages-${pagesize}kB/nr_hugepages
+
+        :param pagesize: hugepages size
+        :param pagenum: page number
+        """
+        pagedir = "%s/hugepages-%skB" % (self.pool_path, pagesize)
+        pagefile = "%s/nr_hugepages" % pagedir
+
+        if not isinstance(int(pagenum), type(0)):
+            raise ValueError("pagenum should be integer, while got "
+                             "%s" % pagenum)
+        if not os.path.isdir(pagedir) or not os.path.isfile(pagefile):
+            raise ValueError("Root hugepages control sysfs directory %s or"
+                             " file %s did not exist" % (pagedir, pagefile))
+        process.system("echo %s > %s" % (pagenum, pagefile), shell=True)
+
+    def get_free_hugepages(self, node, pagesize):
```

Comment:
test_setup already have this API as `get_node_num_huge_pages(self, node, pagesize)`